### PR TITLE
SONARIAC-1445 Migrate Sonar IaC to Starlark

### PR DESCRIPTION
**Pull request by olivier-schmitt-sonarsource**

### CODE REVIEW

File: .cirrus.star [hunk 1]
```diff
@@ -1,4 +1,40 @@
+# Check Starlark specs: https://github.com/bazelbuild/starlark/blob/master/spec.md
+# RE
```
Comment (nils-werner-sonarsource) [inline .cirrus.star:2]: RE is dead, long live DevInfra 😆

File: .cirrus.star [hunk 2]
```diff
@@ -1,4 +1,40 @@
...
+    conf = dict()
+    merge_dict(conf, load_features(ctx))
+    merge_dict(conf, env())
+    merge_dict(conf, build_task())
+    merge_dict(conf, build_test_analyze_task())
+    merge_dict(conf, qa_os_win_task())
+    merge_dict(conf, sca_scan_task())
+    merge_dict(conf, qa_plugin_task())
+    merge_dict(conf, qa_ruling_task())
+    merge_dict(conf, promote_task())
```
Comment (nils-werner-sonarsource) [inline .cirrus.star:39]: In general, the structure is clear to me. However, I would expect to provide the project/repository-related values as arguments of the function call.

File: .cirrus/modules/cache.star [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+#
+# Project version cache
+#
+
+
+def project_version_cache():
+    return {
+        "folder": "$PROJECT_VERSION_CACHE_DIR",
+        "fingerprint_script": "echo $BUILD_NUMBER",
+        "populate_script": "mkdir -p \"$PROJECT_VERSION_CACHE_DIR\"\ntouch \"$PROJECT_VERSION_CACHE_DIR\"/evaluated_project_version.txt\n",
```
Comment (mstachniuk) [inline .cirrus/modules/cache.star:88]: 1. Is it possible to split the lines for better readability?
2. The last `\n` is not needed.

File: STARLARK.md [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+load(".cirrus/modules/base.star", "base_all_tasks")
+load(".cirrus/modules/build.star", "build_all_tasks")
+load(".cirrus/modules/deploy.star", "deploy_all_tasks")
+load(".cirrus/modules/qa.star", "qa_all_tasks")
+load(".cirrus/modules/housekeeping.star", "housekeeping_all_tasks")
+
+
+def main(ctx):
+    conf = {}
+    re_builtins_conf = load_features(ctx)
```
Comment (mstachniuk) [inline STARLARK.md:253]: Is this line needed? It looks like `re_builtins_conf` is not used later?

File: STARLARK.md [hunk 5]
```diff
@@ -0,0 +1,547 @@
...
+load("github.com/SonarSource/cirrus-modules@v2", "load_features")
+load(".cirrus/modules/base.star", "base_all_tasks")
+load(".cirrus/modules/build.star", "build_all_tasks")
+load(".cirrus/modules/deploy.star", "deploy_all_tasks")
+load(".cirrus/modules/qa.star", "qa_all_tasks")
+load(".cirrus/modules/housekeeping.star", "housekeeping_all_tasks")
+
+
+def main(ctx):
+    conf = {}
```
Comment (mstachniuk) [inline STARLARK.md:252]: In code this line is defined as `conf = dict()`. This should be consistent.

**Review by mstachniuk — APPROVED** [review]
LGTM

File: STARLARK.md (hunk 4 above)
Comment (olivier-schmitt-sonarsource) [reply STARLARK.md:253]: Indeed!
Good catch!

File: STARLARK.md (hunk 5 above)
Comment (olivier-schmitt-sonarsource) [reply STARLARK.md:252]: Indeed!
Good catch!

File: .cirrus/modules/cache.star (hunk 3 above)
Comment (olivier-schmitt-sonarsource) [reply .cirrus/modules/cache.star:88]: Yes definitely, I did it for the most problematic scripts.

File: .cirrus.star (hunk 2 above)
Comment (olivier-schmitt-sonarsource) [reply .cirrus.star:39]: I don't understand what you mean.
Can you provide an example, please?

**Pull request merged**